### DISCUSSION TITLE
no main manifest attribute, in Greengrass-1.0-SNAPSHOT.jar

### DISCUSSION THREAD

**Discussion by SatyaaDixit** [aws/aws-iot-device-sdk-java-v2#448] [Q&A] [answered]
Hello Support,

While running the greengrass.BasicDiscovery having some issues so was trying to run/compile the file in the local system itself but it landing to another issue which is below mentioned. Requesting your support on the same.

Command Using
1. 
java -jar Greengrass-1.0-SNAPSHOT.jar -Dexec.mainClass=greengrass.BasicDiscovery -Dexec.args='--cert /home/admin/certs/device.pem.crt --key /home/admin/certs/private.pem.key --ca_file /home/admin/certs/AmazonRootCA1.pem --thing_name MyGreengrassCore --region us-gov-west-1 --topic clients/ggv2client/hello/world --proxy.port <port> --proxy.host =<IP>'

2. java -cp Greengrass-1.0-SNAPSHOT.jar greengrass.BasicDiscovery -Dexec.args='--cert /home/admin/certs/device.pem.crt --key /home/admin/certs/private.pem.key --ca_file /home/admin/certs/AmazonRootCA1.pem --thing_name MyGreengrassCore --region us-gov-west-1 --topic clients/ggv2client/hello/world --proxy.port <port> --proxy.host =<IP>'

Thanks,
Satya

**Comment by SatyaaDixit** [comment]
To compile the code used follow.
mvn compile exec:java -pl samples/Greengrass -Dexec.mainClass=greengrass.BasicDiscovery -Dexec.args='--cert /home/admin/certs/device.pem.crt -ey /home/admin/certs/private.pem.key --ca_file /home/admin/certs/AmazonRootCA1.pem --thing_name MyGreengrassCore --region us-gov-west-1 --topic clients/MyGreengrassCore/hello/world --proxy.port <port>--proxy.host =<IP>'

mvn clean install exec:java -pl samples/Greengrass -Dexec.mainClass=greengrass.BasicDiscovery -Dexec.args='--cert /home/admin/certs/device.pem.crt --key /home/admin/certs/private.pem.key --ca_file /home/admin/certs/AmazonRootCA1.pem --thing_name MyGreengrassCore --region us-gov-west-1 --topic clients/ggv2client/hello/world --proxy.port <port>--proxy.host =<IP>'

**Comment by yasminetalby** [comment, marked as answer]
Hello @SatyaaDixit ,

Thank you very much for your submission. 
Could you specify what issue you are running into. In your submission you mention "_having some issues so was trying to run/compile the file in the local system itself but it landing to another issue which is below mentioned_" without providing further information besides the command used (I am assuming that one of the two issue is the "no main manifest attribute"). I was wondering if you could perhaps provide both full error message you are running into using the sample?


I am assuming you have already reviewed this but here is the general sample guidance documentation to run the sample and retrieve logging information: https://github.com/aws/aws-iot-device-sdk-java-v2/blob/main/samples/README.md
This documentation might also be useful to troubleshoot any issue with building the Greengrass basic discovery sample: https://docs.aws.amazon.com/greengrass/v2/developerguide/troubleshooting-client-devices.html#greengrass-discovery-issues

Assuming you have installed the SDK following the readme [documentation](https://github.com/aws/aws-iot-device-sdk-java-v2), you should be able to run the sample as follow : `mvn compile exec:java -pl samples/Greengrass -Daws.crt.debugnative=true -Daws.crt.log.level=Debug -Daws.crt.log.destionation=Stdout -Dexec.mainClass=pubsub.PubSub -Dexec.args='--endpoint <endpoint> --cert <path to cert> --key <path to key> --ca_file <path to ca file>'` 

However, note that this sample is to be used alongside tutorials shared in the readme.md file : https://github.com/aws/aws-iot-device-sdk-java-v2/blob/main/samples/Greengrass/README.md

Best regards,

Yasmine
